.buildkite/scripts/steps/integration_tests.sh

```diff
@@ -20,6 +20,8 @@ AGENT_PACKAGE_VERSION="${OVERRIDE_AGENT_PACKAGE_VERSION}" DEV=true EXTERNAL=true
 set +e
 AGENT_VERSION="${OVERRIDE_TEST_AGENT_VERSION}" TEST_INTEG_CLEAN_ON_EXIT=true SNAPSHOT=true mage integration:test
 TESTS_EXIT_STATUS=$?
+AGENT_VERSION="${OVERRIDE_TEST_AGENT_VERSION}" STACK_PROVISIONER="ess" TEST_INTEG_CLEAN_ON_EXIT=true SNAPSHOT=true mage integration:test
```

Comment:
Hmm, I'm not sure I want to run every integration test twice. That is going to cause the tests to take something like 2.5 hours since it will double the run time.

I would only run a basic sanity check, probably only https://github.com/elastic/elastic-agent/blob/main/testing/integration/monitoring_logs_test.go is enough. 

There aren't any actual changes in agent itself for serverless, running the other tests is interesting but it isn't telling us anything about agent, only Fleet.

Reply:
We could consider moving this to a separate parallel build step. That will speed it up but I still wouldn't want to run all the tests twice.

https://github.com/elastic/elastic-agent/blob/123ba9ce80c9865f72fa3659b5cafe9b51954f49/.buildkite/pipeline.yml#L34-L42

Just duplicating that pipeline and parameterizing this script to take the provisioner type and the mage target to run would work.

Also if you used `ess` to abbreviate `Elasticsearch Serverless` then that is confusing because that abbreviation already means `Elasticsearch service` which is a common name for the stateful cloud offering. We should rename it to something more obvious.

Reply:
The script also builds the agent, so just running it twice might just slow it down? Unless we refactor the pipeline to do building as a separate step.

I fear we're gonna get an appreciable amount of slowdown no matter what, though. In my experience bringing up the serverless cluster, and waiting for it to return healthy (specifically, waiting for DNS to resolve) takes a non-zero amount of time.

Perhaps we can run some of the pipelines in parallel? At that point we're kind of the outside the scope of this PR, though.